code/modules/modular_computers/file_system/programs/configurator.dm

```diff
@@ -19,20 +19,16 @@
 	if(!computer)
 		return 0
 
-	var/obj/item/computer_hardware/battery/battery_module = computer.all_components[MC_CELL]
-
 	var/list/data = get_header_data()
 
 	data["disk_size"] = computer.max_capacity
 	data["disk_used"] = computer.used_capacity
 	data["power_usage"] = computer.last_power_usage
-	data["battery_exists"] = battery_module ? 1 : 0
-	if(battery_module?.battery)
-		data["battery_rating"] = battery_module.battery.maxcharge
-		data["battery_percent"] = round(battery_module.battery.percent())
-
-	if(battery_module?.battery)
-		data["battery"] = list("max" = battery_module.battery.maxcharge, "charge" = round(battery_module.battery.charge))
+	if(computer.internal_cell)
+		data["battery"] = list(
+			"max" = computer.internal_cell.maxcharge,
+			"charge" = round(computer.internal_cell.charge),
+		)
```

Comment:
If ui_data doesn't return a key `battery` in the list, tgui will retain the previous value, therefore it is necessary for it to always have some default value.

```suggestion
	data["battery"] = null
	if(computer.internal_cell)
		data["battery"] = list(
			"max" = computer.internal_cell.maxcharge,
			"charge" = round(computer.internal_cell.charge),
		)
```